File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/single_child_scroll_view.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/enums.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// SingleChildScrollView — torna o filho rolável num eixo.
Widget buildSingleChildScrollView(
    BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return SingleChildScrollView(
    // Default do Flutter é vertical (≠ Wrap.direction, que é horizontal).
    scrollDirection: axisFrom(p['scrollDirection'] ?? 'vertical'),
    reverse: p['reverse'] as bool? ?? false,
    padding: parseEdgeInsets(p['padding']),
    child: r.maybeRender(context, node.child),
  );
}

File: docs/WidgetMill/packages/spec/src/nodes/singleChildScrollView.ts
import { z } from "zod";
import { EdgeInsets } from "../complex/edgeInsets";
import { AXIS } from "../enums";
import { NodeRef } from "../node-ref";

/** SingleChildScrollView — torna o filho rolável num eixo (catálogo Flutter). */
export const SingleChildScrollViewNode = z.object({
  type: z.literal("singleChildScrollView"),
  props: z
    .object({
      scrollDirection: z.enum(AXIS).optional(),
      reverse: z.boolean().optional(),
      padding: EdgeInsets.optional(),
    })
    .default({}),
  child: NodeRef.optional(),
});
